gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -414,6 +415,9 @@ public void apply(JobListener jobListener, JobContext jobContext)
         workUnitsCreationTimer.stop(this.eventMetadataGenerator.getMetadata(this.jobContext,
             EventName.WORK_UNITS_CREATION));
 
+        this.jobContext.getJobState().setProp(
+            MetricNames.NUM_WORKUNITS, workUnitStream.getMaterializedWorkUnitCollection().size());
```

Comment:
On second reading - Is numWorkunits a CodaHale metric that gets emitted via MetricReporter? I think the metrics listed in MetricNames are intended for the reporter metrics. If numWorkUnits is intended for a different purpose, let's not add it into MetricNames class.

Reply:
Right, it's not a codahale metric name. Can you suggest a right place? The current class (AbstractJobLauncher) ?